tables/Tables/TableUtil.cc

```diff
@@ -288,26 +292,23 @@ namespace casacore {
       // Check that no empty parts are given.
       if (anyEQ (names, String())) {
         msg = "empty name part given";
-      } else if (names.size() > 1  ||  mustExist) {
-        // No need to check if new table and no subtable given.
-        // Check if main table exists.
+      } else if (names.size() > 1) {
+        // Subtable given; check if previous parts exist.
+        // First check if main table exists.
         if (! Table::isReadable (names[0])) {
           msg = "main table " + names[0] + " does not exist";
         } else {
           tab = Table(names[0]);
           // Get name of last subtable.
-          if (names.size() > 1) {
-            lastPart = names[names.size()-1];
-            // Check if all subtables exist, maybe except last one.
-            uInt nrname = mustExist ? names.size() : names.size()-1;
-            for (uInt i=1; i<nrname; ++i) {
-              if (! tab.keywordSet().isDefined(names[i])) {
-                msg = "subtable " + names[i] + " is unknown";
-                tab = Table();
-                break;
-              }
-              tab = tab.keywordSet().asTable (names[i]);
+          lastPart = names[names.size()-1];
+          // Check if all subtables exist, except last one.
+          for (uInt i=1; i<names.size()-1; ++i) {
+            if (! tab.keywordSet().isDefined(names[i])) {
+              msg = "subtable " + names[i] + " is unknown";
+              tab = Table();
```

Comment:
Optional: Resetting `tab` isn't strictly necessary, since you're throwing an exception below. If it is necessary, I would reset `tab` just before throwing the exception, so it's always reset in all error cases.